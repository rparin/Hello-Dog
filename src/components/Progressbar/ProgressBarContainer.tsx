import cn from "@/utils/cn";
import ProgressbarItem from "@/components/Progressbar/ProgressbarItem";
import { DogInfoItem } from "@/schema/DogBreedInfo";

export default function ProgressbarContainer({
  className,
  title,
  scales,
}: {
  className?: string;
  title: string;
  scales: Array<DogInfoItem>;
}) {
  const barItems = scales.map((item) => {
    return (
      <ProgressbarItem
        className="md:pb-1"
        key={item.name}
        barInfo={item}></ProgressbarItem>
    );
  });

  return (
    <div
      className={cn(
        "flex h-full w-full flex-col items-center bg-accent text-black",
        className
      )}>
      <h2 className="text-base leading-none md:text-xl">{title}</h2>
      <ul
        tabIndex={0}
        aria-label={`${title}`}
        className="hide-scrollbar flex h-full w-full flex-col overflow-auto">
        {barItems}
      </ul>
    </div>
  );
}
