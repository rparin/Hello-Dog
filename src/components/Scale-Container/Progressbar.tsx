import { cn } from "@/lib/utils";

export type barType = {
  name: string;
  value: 1 | 2 | 3 | 4 | 5;
};

export default function Progressbar({
  className,
  barValue,
}: {
  className?: string;
  barValue: barType;
}) {
  const scale = {
    1: "w-1/5 bg-[#F03F4B]",
    2: "w-2/5 bg-[#E54C25]",
    3: "w-3/5 bg-[#E59E24]",
    4: "w-4/5 bg-[#9ACA3C]",
    5: "w-full bg-[#55B949]",
  };
  return (
    <div className={cn("flex h-full w-full flex-col", className)}>
      <p className="ml-1.5 mr-1.5 border-b border-solid border-black text-xs leading-none md:text-base">
        {barValue.name}
      </p>
      <div className="mx-1 mt-0.5 h-full rounded-3xl bg-white md:mx-0.5">
        <div
          className={cn(
            "h-full rounded-3xl",
            scale[barValue.value as keyof typeof scale],
          )}
        ></div>
      </div>
    </div>
  );
}
