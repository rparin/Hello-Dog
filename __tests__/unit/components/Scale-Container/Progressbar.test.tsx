import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import Progressbar from "@/components/Scale-Container/Progressbar";

expect.extend(toHaveNoViolations);

test("should have no accessibility violations", async () => {
  const { container } = render(
    <Progressbar barValue={{ name: "Energy", value: 3 }}></Progressbar>
  );
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});

test("it renders component unchanged", () => {
  const { container } = render(
    <Progressbar barValue={{ name: "Energy", value: 3 }}></Progressbar>
  );
  expect(container).toMatchSnapshot();
});
