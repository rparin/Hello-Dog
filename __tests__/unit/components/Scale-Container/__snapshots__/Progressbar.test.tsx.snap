// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <div
    class="flex h-full w-full flex-col"
  >
    <p
      class="ml-1.5 mr-1.5 border-b border-solid border-black text-xs leading-none md:text-base"
    >
      Energy
    </p>
    <div
      class="mx-1 mt-0.5 h-full max-h-3 min-h-2 rounded-3xl bg-white md:mx-0.5 md:max-h-4 lg:max-h-5"
    >
      <div
        class="h-full rounded-3xl w-3/5 bg-[#E59E24]"
      />
    </div>
  </div>
</div>
`;
